### ISSUE TITLE
Revise Assembly support, possibly into a new DLL

### ISSUE THREAD

**Issue by ffhacksterex** [ffhacksterex/FFHacksterEx#109]
The current assembly support is extremely hacky.
It relies on two things:
- markers (formatted end-of-line comments)
- a .dialogue file for the dialogue editor

This means that changes have to be made in multiple places to support assembly changes. Also, some changes that can be made in ROM without hex editing require markers in assembly.
This also means the Disch disassembly1.0 can't be used out-of-the-box.
Let's investigate to see if there's a way (short of writing or embedding an assembler) to enable assembly source changes in the FFHEx editors.

**Comment by ffhacksterex** [comment]
@essellejaye  It's nice to have you back!
Any thoughts on this assembly issue?

**Comment by essellejaye** [comment]
@ffhacksterex  Nice to be back, but I'm not sure how long it'll be.
For this one, hmm....
Not sure that we necessarily need to mingle this with the JSON stuff.
I don't see a way around a rewrite. The open question last time I left was if we keep support for the existing ff1asm-disch.dll and just add a new one.
That might cause some confusion, so maybe a new DLL compatible with Disch 1.0 is the default.
It might be tough making the current DLL work with both Disch 1.0 and Annotated Disch.

Do you envision supporting Disch 1.0 and dropping support for Annotated Disch (which I don't think anyone is using)?
If so,  then maybe we version this change as FFHEx 1.0.0.0 and remove the existing ff1asm-disch.dll?

**Comment by essellejaye** [comment]
@essellejaye Bear in mind that the last proposed change would alter the values file, so any path we take will potentially cause compatibility issues.
I think we all agreed not to expect out-of-the-box support for existing hacks even with support for Disch 1.0.
Depending on the path we take, there might be a requirement to do a lot of updates to the values file to get existing projects to work, so we'll need to talk about this a bit more before starting.
